org/all-prs.ts

```diff
@@ -1,31 +1,72 @@
+  // We return a closure here so that the (promise is resolved|closure is invoked) 
+  // during test time and not when we call rfc().
+  return () => {
+    return closure instanceof Promise ? closure : Promise.resolve(closure())
+  }
+}
+
+// Either schedules the promise for execution via Danger, or invokes closure.
+const runRFC = (reason: string, closure: () => void | Promise<any>) =>
+  closure instanceof Promise ? schedule(closure) : closure()
+
+const rfc: any = isJest ? storeRFC : runRFC
 
 import yarn from "danger-plugin-yarn"
-rfc("1", "Highlight package dependencies on Node projects", () => {
-  schedule(yarn())
+rfc("Highlight package dependencies on Node projects", async () => {
+  await yarn()
 })
 
 import spellcheck from "danger-plugin-spellcheck"
-rfc("2", "Keep our Markdown documents awesome", () => {
-  schedule(spellcheck({ settings: "artsy/artsy-danger@spellcheck.json" }))
+rfc("Keep our Markdown documents awesome", async () => {
+  await spellcheck({ settings: "artsy/artsy-danger@spellcheck.json" })
 })
 
 // https://github.com/artsy/artsy-danger/issues/5
-rfc("5", "No PR is too small to warrant a paragraph or two of summary", () => {
+export const rfc5 = rfc("No PR is too small to warrant a paragraph or two of summary", () => {
+  const pr = danger.github.pr
   if (pr.body.length === 0) {
     fail("Please add a description to your PR.")
   }
 })
 
 // https://github.com/artsy/artsy-danger/issues/13
-rfc("13", "Always ensure we assign someone, so that our Slackbot work correctly", () => {
+export const rfc13 = rfc("Always ensure we assign someone, so that our Slackbot work correctly", () => {
+  const pr = danger.github.pr
   const wipPR = pr.title.includes("WIP ") || pr.title.includes("[WIP]")
   if (!wipPR && pr.assignee === null) {
     warn("Please assign someone to merge this PR, and optionally include people who should review.")
   }
 })
+
+// https://github.com/artsy/artsy-danger/issues/16
+export const rfc16 = rfc("Require changelog entries on PRs with code changes", async () => {
+  const pr = danger.github.pr
+  const getFile = danger.github.utils.fileContents
+  const files = [...danger.git.modified_files, ...danger.git.modified_files]
+  const changelogs = ["CHANGELOG.md", "changelog.md", "CHANGELOG.yml"]
+  let changelogExists = false
+  for (var changelog of changelogs) {
+    const content = await getFile(changelog)
+    if (content) {
+      changelogExists = true;
+    }
+  }
+  if (changelogExists) {
+    const hasCodeChanges = (files.filter(file => {
+      return file.match(/(test|spec)/i) === null
+    }).length > 0)
+    const changelogRegex = new RegExp(changelogs.join('|'), 'i')
+    const hasChangelogChanges = (files.filter(file => {
+      return file.match(changelogRegex) !== null
+    }).length > 0)
+    if (hasCodeChanges && !hasChangelogChanges) {
+      warn("It looks like code was changed without adding anything to the Changelog")
```

Comment:
I changed back to using `fileContents` because it seemed more idiomatic, but happy to discuss!